### PR TITLE
Add -Wb,opendds_sequence_suffix= back to opendds_idl, add test

### PR DESCRIPTION
**Pull request by jwillemsen**
…nother IDL type

    * tests/DCPS/Compiler/idl_test3_lib/FooDef.idl:

### CODE REVIEW

**Comment by jwillemsen** [comment]
Reproducer for #2965

**Comment by jwillemsen** [comment]
Fixed #2965

File: dds/idl/be_global.cpp [hunk 1]
```diff
@@ -348,6 +348,9 @@ BE_GlobalData::parse_args(long& i, char** av)
   static const char EXPORT_FLAG[] = "--export=";
   static const size_t EXPORT_FLAG_SIZE = sizeof(EXPORT_FLAG) - 1;
 
+  static const char TYPESEQUENCESUFFIC_FLAG[] = "--typeSequenceSuffix=";
```
Comment (iguessthislldo) [inline dds/idl/be_global.cpp:351]:
```suggestion
  static const char TYPESEQUENCESUFFIC_FLAG[] = "--type-sequence-suffix=";
```

To match the other long options.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: dds/idl/be_global.cpp (hunk 1 above)
Comment (jwillemsen) [reply dds/idl/be_global.cpp]: Changed

File: dds/idl/be_global.cpp [hunk 2]
```diff
@@ -348,6 +348,9 @@ BE_GlobalData::parse_args(long& i, char** av)
   static const char EXPORT_FLAG[] = "--export=";
   static const size_t EXPORT_FLAG_SIZE = sizeof(EXPORT_FLAG) - 1;
 
+  static const char TYPESEQUENCESUFFIC_FLAG[] = "--type-sequence-suffix=";
```
Comment (iguessthislldo) [inline dds/idl/be_global.cpp:351]:
```suggestion
  static const char TYPESEQUENCESUFFIX_FLAG[] = "--type-sequence-suffix=";
```

Change the other uses of this.

Also add `--type-sequence-suffix` to [`--help`](https://github.com/objectcomputing/OpenDDS/blob/28ce33c4ff01963861abce7a3f77c0ef865c8117/dds/idl/be_util.cpp#L99).

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

**Review by iguessthislldo — APPROVED** [review]

File: dds/idl/be_util.cpp [hunk 3]
```diff
@@ -125,6 +125,7 @@ be_util::usage()
     ACE_TEXT("\t\t\t\t\t(sequential if not set)\n")
     ACE_TEXT(" --default-try-construct\t\tset default XTypes try-construct behavior to ")
     ACE_TEXT("discard, use-default, or trim\n\t\t\t\t\t(discard if not set)\n")
+    ACE_TEXT(" --type-sequence-suffix=<postfix>\tset the implied DDS sequence postfix (default is 'Seq')\n")
```
Comment (mitza-oci) [inline dds/idl/be_util.cpp:128]:
```suggestion
    ACE_TEXT(" --type-sequence-suffix=<postfix>\tset the implied DDS sequence suffix (default is 'Seq')\n")
```

**Review by mitza-oci — CHANGES_REQUESTED** [review]

**Comment by mitza-oci** [comment]
An equivalent option (but under a different name) used to be supported, starting with a011cf5ee7a923a8d11a05f945d25894420eecaf and ending at 9b87c5f4785204369f1280fda66ae84aabd8fcba

Based on this, I wonder if the same name should be used here.  It also follows the pattern of using `-Wb,` for fine tuning and little-used options (like `-X` for the JVM).

**Comment by jwillemsen** [comment]
I am fine with using the option used in the past, will update this PR next week

**Comment by jwillemsen** [comment]
Updated PR to use `-Wb,opendds_sequence_suffix`

**Review by mitza-oci — CHANGES_REQUESTED** [review]
The generated files need to be in `.gitignore`.  Other than that I think it's ready.
https://github.com/objectcomputing/OpenDDS/pull/2966/checks?check_run_id=3583850340

**Review by mitza-oci — APPROVED** [review]

**Pull request merged**